### PR TITLE
[Improvement] Use IO List for each legacy Protocol Request

### PR DESCRIPTION
**Pull request by Argonus**
Continuation of (404)[https://github.com/kafkaex/kafka_ex/pull/404].
FYI @jbruggem

### CODE REVIEW

File: .tool-versions
Comment (joshuawscott) [inline .tool-versions:1]: Can you remove this since we've already added to .gitignore?

**Review by joshuawscott — APPROVED** [review]
Other than the extra .tool-versions file, this LGTM

File: .tool-versions
Comment (Argonus) [reply .tool-versions:1]: 🤦 Yeah.

**Comment by Argonus** [comment]
@joshuawscott done :)

**Pull request merged**